src/Protocol/ActivityPub/Transmitter.php

```diff
@@ -721,6 +701,19 @@ private static function createPermissionBlockForItem($item, $blindcopy, $last_id
 			unset($receivers['bcc']);
 		}
 
+		foreach (['to' => Tag::TO, 'cc' => Tag::CC, 'bcc' => Tag::BCC] as $element => $type) {
+			if (!empty($receivers[$element])) {
+				foreach ($receivers[$element] as $receiver) {
+					if ($receiver == ActivityPub::PUBLIC_COLLECTION) {
+						$name = Receiver::PUBLIC_COLLECTION;
+					} else {
+						$name = trim(parse_url($receiver, PHP_URL_PATH), '/');
+					}
+					Tag::store($item['uri-id'], $type, $name, $receiver);
+				}
+			}
```

Comment:
The `$receivers` array already has all the keys that `$element` can take, so the following `foreach` will naturally skip empty arrays.
```suggestion
			foreach ($receivers[$element] as $receiver) {
				if ($receiver == ActivityPub::PUBLIC_COLLECTION) {
					$name = Receiver::PUBLIC_COLLECTION;
				} else {
					$name = trim(parse_url($receiver, PHP_URL_PATH), '/');
				}

				Tag::store($item['uri-id'], $type, $name, $receiver);
			}
			
```

Reply:
I reverted my change because indeed we remove the content of the `bcc` field. With the proposed change I get these warnings:
```
[19-Feb-2022 20:20:06 UTC] PHP Warning:  Undefined array key "bcc" in /src/Protocol/ActivityPub/Transmitter.php on line 705
[19-Feb-2022 20:20:06 UTC] PHP Warning:  foreach() argument must be of type array|object, null given in /src/Protocol/ActivityPub/Transmitter.php on line 705

```

Reply:
Indeed, thanks for trying!